source/agora/node/Validator.d

```diff
@@ -508,4 +510,22 @@ public class Validator : FullNode, API
 
         return Hash.init;
     }
+
+    /***************************************************************************
+
+        A delegate to be called by Nominator when node's own nomination is
+        invalid
+
+        Params:
+            data = Invalid ConsensusData
+            msg = Reason
+
+    ***************************************************************************/
+
+    private void invalidNominationHandler (const ref ConsensusData data,
+        const ref string msg) @safe
+    {
+        if (msg == "Enrollment: Insufficient number of active validators")
```

Comment:
So if someone accidentally changes the string, then this will no longer work. It's an odd choice. Why not just use an enum?

Reply:
That is why there are comments about being careful about changing the literals :smile: (Ledger.d:683)
Well, changing the return type would probably mean changing the entire `isInvalidReason` family of functions and refactor points they were called. I am not entirely sure if this is worth it.

But I fully agree this puts the weight entirely on our future selfs :smile:

Reply:
`validateConsensusData` now returns an enum. :+1: